engine/src/assets/assets.cpp

```diff
@@ -233,9 +242,12 @@ bool Assets::saveMeta(const AnyAsset& handle) const
             return false;
         }
 
-        auto serializer = core::data::old::JSONSerializer(*stream, 4);
-        serializer.context().push(AssetMeta::Exclude{{"path"}}); // Don't save the path field.
-        serializer.write(*meta, nullptr);
+        nlohmann::json json;
+        for (const auto& [key, value] : meta->params())
+        {
+            json[key] = value;
+        }
+        stream->write(json.dump().c_str(), json.dump().size() * sizeof(char));
```

Comment:
```suggestion
        stream->print(json.dump(4));
```

Missing indentation on the .meta file (for better readability).